src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -2487,8 +2497,9 @@ private void VisitOnStatement(ParseNode node)
 
             if (triggerObject.IsInitialized())
             {
-                AddOpcode(new OpcodePush(triggerObject.VariableNameOldValue));
-                AddOpcode(new OpcodePush(triggerObject.VariableName));
+                // Store the current value into the old value to prep for the first use of the ON trigger:
+                AddOpcode(new OpcodePush(triggerObject.OldValueIdentifier));
+                VisitNode(node.Nodes[1]); // the expression in the on statement.
```

Comment:
Never mind, false alarm.  The act of doing an opcodestore into the remember-old-value variable will effectively do its own eval.